reflex/experimental/vars/base.py

```diff
@@ -0,0 +1,73 @@
+import dataclasses
+import sys
+from typing import Any, Literal, Optional, Type, Union, get_args, get_origin
+from reflex.utils import types
+from reflex.vars import Var, VarData
+
+
+@dataclasses.dataclass(
+    eq=False,
+    frozen=True,
+    **{"slots": True} if sys.version_info >= (3, 10) else {},
+)
+class ImmutableVar(Var):
+    # The name of the var.
+    _var_name: str = dataclasses.field()
+
+    # The type of the var.
+    _var_type: Type = dataclasses.field(default=Any)
+
+    # Whether this is a local javascript variable.
+    _var_is_local: bool = dataclasses.field(default=False)
+
+    # Whether the var is a string literal.
+    _var_is_string: bool = dataclasses.field(default=False)
+
+    # _var_full_name should be prefixed with _var_state
+    _var_full_name_needs_state_prefix: bool = dataclasses.field(default=False)
+
+    # Extra metadata associated with the Var
+    _var_data: Optional[VarData] = dataclasses.field(default=None)
+
+    def get_default_value(self) -> Any:
```

Comment:
i think we can leave this function out here, as this functionality is only used in State classes (and IMO is a bit of a kludge that we should avoid bringing into the new API)

Reply:
should be resolved!